lib/screens/login/login_view_model.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:async';
+
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter/model/login_parameters.dart';
+import 'package:survey_flutter/usecases/base/base_use_case.dart';
+import 'package:survey_flutter/usecases/login_use_case.dart';
+
+final loginViewModelProvider =
+    AsyncNotifierProvider.autoDispose<LoginViewModel, void>(LoginViewModel.new);
+
+class LoginViewModel extends AutoDisposeAsyncNotifier<void> {
+  late LoginUseCase loginUseCase;
+
+  String _email = '';
+  String _password = '';
+
+  final _isValidEmail = StreamController<bool>();
+  Stream<bool> get isValidEmail => _isValidEmail.stream;
+
+  final _isValidPassword = StreamController<bool>();
+  Stream<bool> get isValidPassword => _isValidPassword.stream;
+
+  void checkEmail(String? email) {
+    _email = email ?? '';
+    _isValidEmail
+        .add(!(email == null || email.isEmpty || !email.contains('@')));
+  }
+
+  void checkPassword(String? password) {
+    _password = password ?? '';
+    _isValidPassword
+        .add(!(password == null || password.isEmpty || password.length < 8));
+  }
```

Comment:
modify the methods to return the error message if the value is invalid.
```suggestion
  String? checkEmail(String? email, String errorMessage) {
    _email = email ?? '';
    if (email == null || email.isEmpty || !email.contains('@')) {
      return errorMessage;
    }
    return null;
  }

  String? checkPassword(String? password, String errorMessage) {
    _password = password ?? '';
    if (password == null || password.isEmpty || password.length < 8) {
      return errorMessage;
    }
    return null;
  }
```